internal/terraform/datadir/module_types.go

```diff
@@ -0,0 +1,36 @@
+package datadir
+
+import (
+	"github.com/hashicorp/go-getter"
```

Comment:
I appreciate the convenience aspect of leveraging `go-getter` here directly, but it's worth noting that Terraform unfortunately doesn't use these detectors _directly_ and so its understanding of addresses may differ from go-getter for various reasons, some of which are documented [here](https://github.com/hashicorp/terraform/blob/51b0aee36cc2145f45f5b04051a01eb6eb7be8bf/internal/getmodules/package.go#L7-L36) and [here](https://github.com/hashicorp/terraform/blob/332ea1f233891e45bd274e714ab8e4e11c79dc95/internal/getmodules/getter.go#L13-L28).

All that said it shouldn't differ too much and we only use the detection to improve UX, so the cost of getting it wrong is relatively low and I think it's totally fine to use it for now.

It is likely though that in the long run we will maintain some custom logic which more accurately reflects TF Core implementation and doesn't have any side effects such as making network calls.

TL;DR Ignore the above, the use of go-getter is 👌🏻  in this PR 😄

Reply:
👍🏼 Like you say, we aren't interested in validity here, we're interested in figuring out whether we use the Terraform icon or a folder icon or a git icon.